### ISSUE TITLE
When external library polyfills it breaks Next.js

### ISSUE THREAD

**Issue by luthfib** [vercel/next.js#57242]
### Link to the code that reproduces this issue

https://github.com/luthfib/auth0-nextjs-repro/tree/fetch-issue

### To Reproduce

1. Run intstructions as outlined in https://github.com/luthfib/auth0-nextjs-repro/tree/fetch-issue
2. This will also require get env variables for auth0

### Current vs. Expected behavior

This bug is linked to https://github.com/vercel/next.js/issues/56969, however I was able to fix it by doing the following: 

https://github.com/luthfib/auth0-nextjs-repro/blob/fetch-issue/app/api/auth/%5Bauth0%5D/route.ts

I believe Next.js crashed before I included the following line because of this line the Next.js code... https://github.com/vercel/next.js/blob/dc88b413f8f4e7569aa4cb65f817b9c15493bafe/packages/next/src/server/web/adapter.ts#L216. 

I think either Next.js needs to warn in the documentation that any library that polyfills fetch will break Next.js or the error if the library polyfills should be more clear. The library that polyfills in question is the following: https://github.com/Vermonster/fhir-kit-client/blob/3b36f30f29f5bbeac356feed0db6ce3be3eeb711/lib/http-client.js#L3. 

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.6.0: Fri Sep 15 13:41:28 PDT 2023; root:xnu-8796.141.3.700.8~1/RELEASE_ARM64_T6000
Binaries:
  Node: 18.11.0
  npm: 8.19.2
  Yarn: N/A
  pnpm: 8.9.2
Relevant Packages:
  next: 13.5.6
  eslint-config-next: 13.5.5
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

**Comment by icyJoseph** [comment]
Is this caused by Node's Fetch API rather? 

So, rather, other libraries should check if native Node Fetch API is available, and skip doing polyfills?

**Comment by luthfib** [comment]
Yeah, I believe so. It's technically on other libraries to check and not do polyfills. But Not everyone is ahead of the curve like Next.js, and some are still polyfilling, but mostly I think at the very least some documentation around this gotcha would be nice. Because this behavior which other libraries may not be aware of can cause some really unexpected behavior in Next.js. Took us a while to track down this bug, and I'd love it if others didn't have to experience it, or at the very least maybe they could figure out what was going on faster.
